feat(voyage): add rerank API support

Chesars opened the pull request:
## Title

Add Voyage AI Rerank API Support

## Relevant issues

N/A - New feature request

## Pre-Submission checklist

- [x] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [ ] I have added a screenshot of my new test passing locally 
- [x] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] My PR's scope is as isolated as possible, it only solves 1 specific problem

## Type

🆕 New Feature

## Changes

This PR adds support for **Voyage AI rerank models** to LiteLLM's rerank API.

### Supported Models

| Model | Context Length | Description |
|-------|----------------|-------------|
| `voyage/rerank-2.5` | 32,000 tokens | Best quality, multilingual, instruction-following |
| `voyage/rerank-2.5-lite` | 32,000 tokens | Optimized for latency and cost |
| `voyage/rerank-2` | 16,000 tokens | Legacy model |
| `voyage/rerank-2-lite` | 8,000 tokens | Legacy model |

### Usage Example

**Python SDK:**
```python
from litellm import rerank
import os

os.environ["VOYAGE_API_KEY"] = "your-api-key"

response = rerank(
    model="voyage/rerank-2.5",
    query="What is the capital of France?",
    documents=[
        "Paris is the capital of France.",
        "London is the capital of England.",
        "Berlin is the capital of Germany.",
    ],
    top_n=3,
)

print(response.results)
# [
#   {"index": 0, "relevance_score": 0.8867},
#   {"index": 2, "relevance_score": 0.3535},
#   {"index": 1, "relevance_score": 0.3398}
# ]
```

**Async:**
```python
from litellm import arerank

response = await arerank(
    model="voyage/rerank-2.5-lite",
    query="Best programming language?",
    documents=["Python is...", "Rust is...", "Go is..."],
    top_n=2,
)
```

**LiteLLM Proxy (config.yaml):**
```yaml
model_list:
  - model_name: rerank-2.5
    litellm_params:
      model: voyage/rerank-2.5
      api_key: os.environ/VOYAGE_API_KEY
```

### Files Changed

| File | Description |
|------|-------------|
| `litellm/llms/voyage/rerank/transformation.py` | VoyageRerankConfig class with request/response transformation |
| `litellm/llms/voyage/rerank/handler.py` | Handler placeholder (HTTP in llm_http_handler) |
| `litellm/__init__.py` | Export VoyageRerankConfig |
| `litellm/rerank_api/main.py` | Add voyage provider to rerank function |
| `litellm/utils.py` | Register voyage in get_provider_rerank_config |
| `model_prices_and_context_window.json` | Add rerank-2.5 and rerank-2.5-lite models |
| `docs/my-website/docs/providers/voyage.md` | Documentation for Voyage rerank |
| `docs/my-website/docs/rerank.md` | Add Voyage to supported providers |
| `tests/test_litellm/llms/voyage/rerank/` | 18 unit tests |

### Tests added

Chesars committed:
feat(voyage): add rerank API support

Add support for Voyage AI rerank models (rerank-2.5, rerank-2.5-lite,
rerank-2, rerank-2-lite) to the LiteLLM rerank API.

Changes:
- Add VoyageRerankConfig transformation class
- Register voyage provider in rerank_api/main.py
- Add voyage case in utils.py get_provider_rerank_config
- Add rerank-2.5 and rerank-2.5-lite models to pricing JSON
- Add unit tests for transformation logic
- Update documentation for voyage.md and rerank.md

Usage:
```python
from litellm import rerank

response = rerank(
    model="voyage/rerank-2.5",
    query="What is the capital of France?",
    documents=["Paris is...", "London is..."],
    top_n=3,
)
```

Chesars committed:
refactor(voyage): simplify rerank transformation code

Remove verbose docstrings to align with other providers (jina_ai pattern).
No functional changes - 168 lines vs 169 for jina_ai.

krrishdholakia commented on model_prices_and_context_window.json:
```diff
@@ -26770,6 +26770,28 @@
         "output_cost_per_token": 0.0
     },
+    "voyage/rerank-2.5": {
+        "input_cost_per_query": 5e-08,
```
i'm confused which of the ways is it?

is it cost per query or cost per token? @Chesars

Chesars replied:
The cost is per token. I added input_cost_per_query because the existing rerank-2 and rerank-2-lite models in the JSON had input_cost_per_query set to the same value as  input_cost_per_token. However, this is technically incorrect since Voyage AI charges per token, not per query. Thanks for catching that! Pushed a fix: https://github.com/Chesars/litellm/commit/4392fce642

Chesars committed:
fix(voyage): remove incorrect input_cost_per_query from rerank models

Voyage AI charges per token, not per query. The input_cost_per_query
field was incorrectly set to the same value as input_cost_per_token
in the existing rerank-2 and rerank-2-lite models.

Removes input_cost_per_query from all Voyage rerank models:
- voyage/rerank-2
- voyage/rerank-2-lite
- voyage/rerank-2.5
- voyage/rerank-2.5-lite

Pricing source: https://docs.voyageai.com/docs/pricing
